File: blind75/data_structures/contains_duplicate.py
# https://leetcode-cn.com/problems/contains-duplicate/
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:  # return if there is dup in nums
        if not nums:
            return False
        visited = set()
        for n in nums:
            if n in visited:
                return True
            visited.add(n)
        return False

File: blind75/hashmap/contains_duplicate.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {
	visited := map[int]struct{}{}
	for _, v := range nums {
		a, b := visited[v]
		fmt.Printf("%+v %v\n", a, b)
		if _, has := visited[v]; has {
			return true
		}
		visited[v] = struct{}{}
	}
	return false
}

// func main() {
// 	containsDuplicate([]int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2})
// }
